lib/artirix_data_models/adm_registry.rb

```diff
@@ -1,22 +1,36 @@
-class ArtirixDataModels::DAORegistry
+class ArtirixDataModels::ADMRegistry
+  @instance_mutex = Mutex.new
+
+  def self.instance_mutex
+    ArtirixDataModels::ADMRegistry.instance_variable_get :@instance_mutex
```

Comment:
so with this I know that any subclass of the registry will use the same mutex instance